src/modules/launcher/Wox.Infrastructure/Storage/ListRepository.cs

```diff
@@ -0,0 +1,68 @@
+﻿using NLog.Filters;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Wox.Infrastructure;
+
+namespace Wox.Infrastructure.Storage
+{
+    /// <summary>
+    /// The intent of this class is to provide a basic subset of 'list' like operations, without exposing callers to the internal representation
+    /// of the data structure.  Currently this is implemented as a list for it's simplicity. 
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class ListRepository<T> : IRepository<T>, IEnumerable<T>

```

Comment:
Can we make this an `abstract` class since it is not being instantiated directly?

Reply:
Yep

Reply:
@somil55 I left this as is to facilitate unit testing.